### ISSUE TITLE
Unhandled Exception: type 'Future<Map<dynamic, dynamic>?>' is not a subtype of type 'FutureOr<Map<dynamic, dynamic>>' in type cast

### ISSUE THREAD

**Issue by kightsonsanom** [transistorsoft/flutter_background_geolocation#553]
## Description
  Cast type exception thrown when trying to get device info. The exact place where the cast is performed is line 34 and 35 in `device_info.dart` file : 
```
   Map map = await (_methodChannel.invokeMethod<Map<dynamic, dynamic>>(
          'getDeviceInfo') as FutureOr<Map<dynamic, dynamic>>);
```

## Your Environment
* Plugin version: 4.0.0-nullsafety.5
* Platform: both
* OS version: all
* Device manufacturer / model: all
* Flutter info (`flutter doctor`):
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.0.5, on macOS 11.2.3 20D91 darwin-x64, locale en-PL)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio
[✓] Android Studio (version 4.1)
[✓] IntelliJ IDEA Community Edition (version 2020.3.2)
[✓] Connected device (3 available)

• No issues found!
```

## Expected Behavior
No cast exception thrown

## Actual Behavior
Cast exception is thrown

## Steps to Reproduce and Context
Use `DeviceInfo.getInstance()` method

## Debug logs

``` 
[VERBOSE-2:ui_dart_state.cc(186)] Unhandled Exception: type 'Future<Map<dynamic, dynamic>?>' is not a subtype of type 'FutureOr<Map<dynamic, dynamic>>' in type cast
#0      DeviceInfo.getInstance (package:flutter_background_geolocation/models/device_info.dart:35:28)
#1      TransistorDataService._configureTransistorConsoleLoggingIfNeeded (package:when_place/data/eta/services/transistor_data_service.dart:100:74)
<asynchronous suspension>
#2      TransistorDataService.enableDiagnosticsMode (package:when_place/data/eta/services/transistor_data_service.dart:77:5)
<asynchronous suspension>
#3      EtaEmiter._updateDiagnosticMode (package:when_place/data/eta/eta_emiter.dart:158:11)
<asynchronous suspension>
#4      EtaEmiter._updateTracking (package:when_place/data/eta/eta_emiter.dart:134:5)
<asynchronous suspension>

```

**Comment by christocracy** [comment]
Try testing this fix from Git:

```yaml
dependencies:
  flutter_background_geolocation:
    git:
      url: https://github.com/transistorsoft/flutter_background_geolocation.git
      ref: nullsafety-DeviceInfo
```

**Comment by kightsonsanom** [comment]
I confirm that this works properly with this fix

**Comment by christocracy** [comment]
Released to `4.0.0-nullsafety.6`
